test/functional/ui/screen.lua

```diff
@@ -217,20 +217,38 @@ end
 
 function Screen:wait(check, timeout)
   local err, checked = false
+  local success_seen = false
+  local failure_after_success = false
   local function notification_cb(method, args)
     assert(method == 'redraw')
     self:_redraw(args)
     err = check()
     checked = true
     if not err then
+      success_seen = true
       stop()
+    elseif success_seen then
```

Comment:
Will this trigger if two successes are received? Maybe it should be:

```lua
if success_seen then
  changes_after_success = true
else if not err then
  success_seen = true
  stop()
end
```

and the check:

```lua
if changes_after_success then
  print(...
```